tests/unit/cbprintf/main.c

```diff
@@ -220,22 +240,52 @@ static int rawprf(const char *format, ...)
 #if USE_PACKAGED
 	rv = cbvprintf_package(packaged, sizeof(packaged), format, ap);
 	if (rv >= 0) {
-		rv = cbpprintf(out, NULL, packaged);
+		rv = cbpprintf(out, &outbuf, packaged);
 	}
 #else
-	rv = cbvprintf(out, NULL, format, ap);
+	rv = cbvprintf(out, &outbuf, format, ap);
 #endif
 	va_end(ap);
 
 	if (IS_ENABLED(CONFIG_CBPRINTF_NANO)
 	    && !IS_ENABLED(CONFIG_CBPRINTF_LIBC_SUBSTS)) {
 		zassert_equal(rv, 0, NULL);
-		rv = bp - buf;
+		rv = outbuf.idx;
 	}
 	return rv;
 }
 
-#define TEST_PRF(_fmt, ...) prf(WRAP_FMT(_fmt), PASS_ARG(__VA_ARGS__))
+#define TEST_PRF2(rc, _fmt, ...) do { \
+	char _buf[512]; \
+	char *sp_buf = NULL; /* static package buffer */\
+	if (USE_PACKAGED && !CBPRINTF_MUST_RUNTIME_PACKAGE(0, _fmt, __VA_ARGS__)) { \
+		int rv = 0; \
+		size_t _len; \
+		struct out_buffer package_buf = { \
+			.buf = _buf, .size = ARRAY_SIZE(_buf), .idx = 0 \
+		}; \
+		CBPRINTF_STATIC_PACKAGE(NULL, &_len, _fmt, __VA_ARGS__); \
+		uint8_t __aligned(CBPRINTF_STATIC_PACKAGE_ALIGNMENT) package[_len]; \
+		CBPRINTF_STATIC_PACKAGE(package, &_len, _fmt, __VA_ARGS__); \
+		rv = cbpprintf(out, &package_buf, package); \
+		if (rv >= 0) { \
+			sp_buf = _buf; \
+		} \
+	} \
+	rc = prf(sp_buf, _fmt, __VA_ARGS__); \
+} while (0)
+
+#define TEST_PRF(rc, _fmt, ...) \
```

Comment:
I'm really not a fan of reference parameters in macros.

Reply:
which one? `rc` is passed by value